content/blog/open-sourcing-workshops-on-demand-part-5-create-a-workshop.md

```diff
@@ -0,0 +1,226 @@
+
+![](/img/wod-blogserie3-archi3.png "WOD Overview.")
+
+I﻿ will consider two scenarios going further. In the first one, I will create a simple workshop that does not require any infrastructure but the JupyterHub itself. Then, in a second phase, I will go through a more complex workshop creation process that will cover most of the possible cases I could think of.
+
+# S﻿imple workshop example:
+
+l﻿et's consider that I plan to create a new workshop on the Go language. It becomes more and more popular among the developer community I interact with and one of the the developer was kind enough to agree with working with me on creating this new workshop. After a first meeting, where I explainied to him the creation process, and the expectations, we quickly started to work together. We defined title, abstract,  notebooks' folder name, and student range. As far as infrastructure's requirements, a new kernel was needed. No additional scripts are requiered for this workshop.
+
+A﻿s an admin of the Workshops-on-demand infrastructure, I had to perform several tasks:
+
+##### O﻿n the backend server:
+
+1. ###### Test and validate installation of the new kernel on the staging backend server by:
+
+* * Creating a new branch for this test
+
+  * M﻿odifying the [backend server installation yaml file ](https://github.com/Workshops-on-Demand/wod-backend/blob/main/ansible/install_backend.yml#L326)to include the new kernel.
+
+  * Validating the changes by testing a new backend install process.
+
+  * Pushing the changes to the github repo.
+
+2. ###### Creating a user for the workshop developer on the test/dev and staging backend servers.
+3. ###### Providing to the developer the necessary information to connect to the test/dev and staging backend servers.
+4. ###### Copy over the workshop developer's home folder a workshop template containing examples of introduction, conclusion and lab notebooks, allowing him to start his work.
+5. **Give the developer the wod-notebook repo url for him to fork the repo and work locally on his machine (when workshop does not require appliance but just a jupyter kernel for instance)**
+6. **Whenever ready, a pull request can be made. The admin can then review and accept it. The admin can then perform the necessary steps to prepare the infrastructure to host the workshop**
+
+
+
+##### O﻿n the database server:
+
+In order to exist, a workshop requires serveral entries in the database:
+
+I﻿n the **Workshops table:**
+
+A﻿ new entry will need the following:
+
+![](/img/wod-db-go-1.png "Workshop's fields in the Database.")
+
+* **An id:** A workshop id to be used by backend server automation and Replays table to reference the associated replay video of the workshop.
```

Comment:
In the following set of bullets, be consistent with whether you end them with a period or not as well as the capitalization and punctuation. Generally, it's best to maintain a parallel structure. For example:

An ID: A workshop ID that is to be used by.... (you can either use a period or not at the end)
A name: The workshop's name... (again, use a period or not, but be consistent)
A name of the folder: This contains all the workshops notebooks (consistent punctuation at the end)
A description / abstract: (Define)
A capacity:... (what you have here is fine as is the next one...)

Reset and Idap entries: These are to be used by....

You get the picture :-)